detectable/src/main/java/com/synopsys/integration/detectable/detectables/gradle/inspection/GradleInspectorExtractor.java

```diff
@@ -93,4 +96,30 @@ private Optional<NameVersion> parseRootProjectMetadataFile(File rootProjectMetad
         }
     }
 
+    private File[] sortFilesByDepth(File[] files) {
+        Arrays.sort(files, new Comparator<File>() {
+            @Override
+            public int compare(File o1, File o2) {
+                int n1 = extractDepthNumber(o1.getName());
+                int n2 = extractDepthNumber(o2.getName());
+                return n1 - n2;
+            }
+
+            private int extractDepthNumber(String name) {
+                int i;
+                try {
+                    int s = name.indexOf("depth") + 5;
+                    int e = name.indexOf("_dependencyGraph");
+                    String number = name.substring(s, e);
+                    i = Integer.parseInt(number);
+                } catch(Exception e) {
+                    i = 0; //  default to 0
```

Comment:
This comparator is assuming the file name format is the updated one with depth information in it.

Reply:
I would log an error here. It's quite unlikely to make noise, and would be good to expose so we're not just silencing potential exceptions.

Reply:
You mean fail Detect in that case? If yes, I am not certain we should do that for this kind of Exception. I would just log the exception move on with the next file.

Reply:
Yup, `logger.error()` sounds good.